### DISCUSSION TITLE
Filter startsWith

### DISCUSSION THREAD

**Discussion by Mister-Ritom** [pocketbase/pocketbase#3075] [Q&A]
Is there a way to get filter all records with a startsWith method.
I want to implement a search feature for the users

**Comment by ganigeorgiev** [comment]
I'm not sure I follow.  If you want to filter records that starts with a specific field value, you can try with the `~` operator combined with `%` for the wildcard matching part:

```js
await pb.collection("example").getFullList({
    filter: `name ~ "yourPrefix%"`
})
```

**Comment by Avashist1998** [comment]
I having trouble implementing this prefix search.  I am able to perform the search using the following query but it uses a wildcard as I am trying to geolocate the data.


Working
```curl --location 'https://url/api/collections/collection/records?page=1&perPage=50&filter=(geohash%20~%20%279vk1%27)'```


Not Working
```curl --location 'https://url/api/collections/collection/records?page=1&perPage=50&filter=(geohash~%279vk1%%27)'```

When I run the postman request, I am getting all the records in my collection and the filter is not even applied to the search. 

But the % work in the admin portal, any help would be greatly appreciated



![Screenshot 2025-06-03 at 9 11 41 PM](https://github.com/user-attachments/assets/681d4064-0b8e-4223-9559-c51bda0d8b3f)

**Reply by Avashist1998** [reply]
Sorry for the comment any one curious, one has to encode special character when using them in HTTP request. `%` being one of these special characters.

Working request
```curl --location 'https://url/api/collections/collection/records?page=1&perPage=50&filter=(geohash~%279vk1%25%27)'```
